src/ride/track.c

```diff
@@ -4880,15 +4880,162 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
 	return;
 }
 
+uint32 loc_6CDE51() {
+	// mov     ebx, 80000000h
+	return 0x80000000;
+}
+
+uint32 loc_6CDE4C() {
+	// pop     edi
+	// pop     edx
+	// pop     ecx
+	// pop     ebx
+	// pop     eax
+	return loc_6CDE51();
+}
+
+uint32 loc_6CDE43() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = 0;
+	
+	return loc_6CDE51();
+}
+
+uint32 loc_6CDE38() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = 954; // empty string?
+	
+	return loc_6CDE51();
+}
+
+uint32 loc_6CDE2D() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
+	
+	return loc_6CDE51();
+}
+
+uint32 loc_6CDE22() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_RIDE_CANT_BUILD_THIS_UNDERWATER;
+	
+	return loc_6CDE4C();
+}
+
+uint32 loc_6CDE17() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_CAN_ONLY_BUILD_THIS_ABOVE_GROUND;
+	
+	return loc_6CDE4C();
+}
+
+uint32 loc_6CD920(int *eax, int *ebx, int *ecx, int *edx, int *esi, int *edi, int *ebp) {
+
+	// test    di, 0Fh
+	// jnz     loc_6CDE38
+	if (((*edi & 0xFFFF) & 0xF) != 0) {
+		loc_6CDE38();
+	}
+	
+	// test    bl, 1
+	// jz      short loc_6CD954
+	if (((*ebx & 0xFF) & 0x1) == 0) {
+		loc_6CD954();
+	}
+	
+	// test    bl, 8
+	// jnz     short loc_6CD954
+	if (((*ebx & 0xFF) & 0x8) != 0) {
+		loc_6CD954();
+	}
+	
+	// push    edx
+	
+	// mov     dx, di
+	//dx = di; // EDX = EDX with DX set to zero, then add DI
+	uint32 temp_edx = (*edx & 0xFFFF0000) + (*edi & 0xFFFF);
+	
+	// call    footpath_remove_litter
+	footpath_remove_litter(*eax, *ecx, temp_edx);
+	
+	// push    eax
+	// push    ebx
+	// push    ecx
+	
+	// and     ax, 0FFE0h
+	uint32 temp_x = (*eax & 0xFFFF0000) + (*eax & 0x0FFE0);
+	
+	// and     cx, 0FFE0h
+	uint32 temp_y = (*ecx & 0xFFFF0000) + (*ecx & 0x0FFE0);
+	
+	// mov     bh, 20h ; ' '
+	uint32 temp_ebx = (*ebx & 0xFFFFFF00FF) + (0x20 << 8);
+	
+	// call    map_remove_walls_at
+	map_remove_walls_at(temp_x, temp_y, 0, 0); // What registers are z1/z0?
+	
+	// pop     ecx
+	// pop     ebx
+	// pop     eax
+	// pop     edx
+}
+
+int do_game_command_set_maze_track(int *eax, int *ebx, int *ecx, int *edx, int *esi, int *edi, int *ebp) {
```

Comment:
From https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/windows/maze_construction.c#L495 you can see that it doesn't expect any of the variables to change so you can change these all from pointers (apart from ebx which you have already accounted for which is used for price `money32` and to indicate failure). 

Here is a list of all of the variables used. (note `esi` is never used as its just the type of game command). 
```
ax = x,
bl = flags,
bh = direction,
cx = y,
dl = rideIndex,
dh = mode,
di = z
```